### PR TITLE
Handled null repo, the case where a user deletes the repo referenced in ...

### LINKED ISSUE #16: Deleted branch referenced by open pull request causes failure
One of our devs accidentally deleted a branch that was the source for an open pull request.  After that, the cronjob running ./bin/lander-check-pulls failed because json["head"]["repo"] in the code below was None and you can't get ["clone_url"] from None.

The code should be robust enough to handle this situation.

From src/jenkinsgithublander/github.py:

def make_pull_request_info(json):
    """Wrap pull request json in PullRequestInfo object"""
    return PullRequestInfo(
        number=json["number"],
        base_ref=json["base"]["ref"],
        base_user=json["base"]["user"],
        head_ref=json["head"]["ref"],
        head_sha=json["head"]["sha"],
        head_repo_url=json["head"]["repo"]["clone_url"],
        comments_href=json["_links"]["comments"]["href"],
    )

### PR DESCRIPTION
**Pull request by bac**
...the pull request.  Fixes #16.

### CODE REVIEW

**Comment by jujugui** [comment]
Test FAILed.
Refer to this link for build results: http://ci.jujugui.org:8080/job/jenkins-github-lander/37/

**Comment by jujugui** [comment]
Test PASSed.
Refer to this link for build results: http://ci.jujugui.org:8080/job/jenkins-github-lander/38/

File: src/jenkinsgithublander/tests/test_github.py [hunk 1]
```diff
@@ -121,32 +122,6 @@ def test_open_pull_requests(self):
         )
 
     @responses.activate
-    def test_no_mergeable_pull_requests(self):
```
Comment (mitechie) [inline src/jenkinsgithublander/tests/test_github.py:124]: can we leave this test? I see the other one is basically the same technically, but the idea that 'everything is peachy but nothing to do' is lost in the test changes as far as intent goes.

**Comment by mitechie** [comment]
:+1: with the note that I'm not sure it's best to remove the generic 'nothing to do' test case.

File: src/jenkinsgithublander/tests/test_github.py (hunk 1 above)
Comment (bac) [inline src/jenkinsgithublander/tests/test_github.py:124]: oh, good catch, that is a mistake.  will restore.

**Comment by jujugui** [comment]
Test PASSed.
Refer to this link for build results: http://ci.jujugui.org:8080/job/jenkins-github-lander/39/

**Comment by jujugui** [comment]
Status: merge request accepted. Url: http://ci.jujugui.org:8080/job/jenkins-github-lander-merge

**Comment by bac** [comment]
:shipit:

**Pull request merged**